src/parser/xasl_generation.c

```diff
@@ -13960,6 +13966,27 @@ pt_to_outlist (PARSER_CONTEXT * parser, PT_NODE * node_list, SELUPD_LIST ** selu
 		      regu->value.dbvalptr = value_list->val;
 		      /* move to next db_value holder */
 		      value_list = value_list->next;
+		      if (node->info.query.q.select.hint & PT_HINT_NO_SUBQUERY_CACHE)
```

Comment:
How about checking it in pt_check_corr_subquery_hash_result_cache()

Reply:
If we apply it as you mentioned, since the function is called recursively, if there is a hint in any query lower than itself (whether it is aptr or dptr), it will prevent itself from using the hash result cache. Is it okay to apply it this way?

Reply:
Since pt_check_corr_subquery_hash_result_cache() is called recursively, it would be a good idea to create a new function.

Reply:
I have understood the intent behind your review and applied it to the code. Please check and confirm.